src/lib/watcher/watcher.ts

```diff
@@ -180,22 +179,25 @@ export async function createWatcher(options: Options): Promise<Watcher> {
     }
     restarting = true
     clearConsole()
-    log.trace('hook', { name: 'beforeWatcherStartOrRestart' })
+    log.info('restarting', change)
 
+    log.trace('hook', { name: 'beforeWatcherStartOrRestart' })
     for (const plugin of plugins) {
       const runnerChanges = await plugin.dev.onBeforeWatcherStartOrRestart?.(change)
       if (runnerChanges) {
         link.updateOptions(runnerChanges)
       }
     }
+
     log.trace('hook', { name: 'beforeWatcherRestart' })
     plugins.forEach((p) => {
       p.dev.onBeforeWatcherRestart?.()
     })
-    log.info('restarting', change)
+
```

Comment:
This makes sure all hooks happen **after** logging `restarting` otherwise the output is always weird. Some stuff happened before logging `restarting`, such as the reflection logs in this case